fairscale/nn/wrap/auto_wrap.py

```diff
@@ -4,23 +4,72 @@
 # LICENSE file in the root directory of this source tree.
 
 import contextlib
-from typing import Any, Callable, Dict, Generator, List, Optional, Tuple
+from typing import Any, Callable, Dict, Generator, Optional, Set, Tuple, cast
 
 import torch.nn as nn
 
-from fairscale.nn.data_parallel.fully_sharded_data_parallel import FullyShardedDataParallel
 from fairscale.nn.misc import checkpoint_wrapper
 
-# Modules that don't wrap.
-FSDP_MODULE_EXCLUDE_WRAP = {nn.ModuleList, nn.ModuleDict}
-# Modules that we don't recurse down to their children.
-FSDP_MODULE_BLOCKLIST = {nn.MultiheadAttention}
+
+def should_wrap_default(
+    should_recurse: bool,
+    module: nn.Module,
+    unwrapped_params: int,
+    # These are customizable for this default policy function.
+    min_num_params: int = int(1e8),
+    blacklist: Optional[Set] = None,
+    exclude_wrap: Optional[Set] = None,
+) -> bool:
+    """Default policy function on with :func:`auto_wrap`.
+
+       Return if a module should be wrapped during :func:`auto_wrap`.
+
+       The first three parameters are used by :func:`auto_wrap`. If
+       you write a custom version of this policy function, your version
+       needs to at least accept the first three parameters and free
+       to do whatever you want in the function.
+
+    Args:
+       should_recurse (bool):
```

Comment:
let's just call this `recurse`, this is what PyTorch calls similar args (e.g., in `nn.Module.parameters(recurse=True)`)

Reply:
yeah, better.